src/d/actor/d_a_wall.cpp

```diff
@@ -4,99 +4,354 @@
+
+    for (int i = 0; i < 3; i++) {
+        mDoMtx_stack_c::multVec(&vertex[i], &vertex[i]);
+    }
+    mTri[0].mA = vertex[0];
+    mTri[0].mB = vertex[1];
+    mTri[0].mC = vertex[2];
+
+    cM3d_CalcPla(&mTri[0].mA, &mTri[0].mB, &mTri[0].mC, &mTri[0].mNormal, &mTri[0].mD);
+
+    vertex[0] = m_tri_vtx[mType][0];
+    vertex[1] = m_tri_vtx[mType][2];
+    vertex[2] = m_tri_vtx[mType][3];
+    for (int i = 0; i < 3; i++) {
+        mDoMtx_stack_c::multVec(&vertex[i], &vertex[i]);
+    }
+    mTri[1].mA = vertex[0];
+    mTri[1].mB = vertex[1];
+    mTri[1].mC = vertex[2];
+    cM3d_CalcPla(&mTri[1].mA, &mTri[1].mB, &mTri[1].mC, &mTri[1].mNormal, &mTri[1].mD);
 }
 
 /* 00000D84-00000F74       .text set_effect__8daWall_cFv */
 void daWall_c::set_effect() {
-    /* Nonmatching */
+    u16 projection_id[3] = {0xa16e, 0xa170, 0xa172};
+    u16 particle_id[3] = {0xa16f, 0xa171, 0xa173};
+
+    csXyz local_28 = current.angle;
+    local_28.y += 0x8000;
+    mpDst = 200.00;
+
+    switch (mType) {
+    case 0:
+    case 1:
+    case 2:
+        dComIfGp_particle_setProjection(projection_id[mType], &current.pos, &current.angle, NULL,
+                                        0xFF, NULL, fopAcM_GetRoomNo(this), &mTevStr.mColorK0, &mTevStr.mColorK0);
+        dComIfGp_particle_setProjection(projection_id[mType], &current.pos, &local_28, NULL,
+                                        0xFF, NULL, fopAcM_GetRoomNo(this), &mTevStr.mColorK0, &mTevStr.mColorK0);
+        mpEmitter = dComIfGp_particle_set(particle_id[mType], &current.pos, &current.angle, NULL,
+                                        mpDst, &mSmokeCb, fopAcM_GetRoomNo(this));
+        if (mpEmitter != NULL) {
+            mpEmitter->mFlags = mpEmitter->mFlags | 0x40;
+        }
+        break;
+    default:
+        mpEmitter = NULL;
+        break;
+    }
+    mBreakCounter = 1;
+    mState = 1;
+    dComIfG_Bgsp()->Release(mpBgW);
+    if (mSwitchIndex != 0xFF)
+        dComIfGs_onSwitch(mSwitchIndex, fopAcM_GetHomeRoomNo(this));
 }
 
 /* 00000F74-00000FE4       .text set_se__8daWall_cFv */
 void daWall_c::set_se() {
-    /* Nonmatching */
+    /* Nonmatching - incorrect rodata offsets */
+    fopAcM_seStart(this, JA_SE_OBJ_BOMB_WALL_BRK, 0);
 }
 
 /* 00000FE4-00001044       .text _draw__8daWall_cFv */
 BOOL daWall_c::_draw() {
-    /* Nonmatching */
+    g_env_light.settingTevStruct(TEV_TYPE_BG0, &current.pos, &mTevStr);
+    g_env_light.setLightTevColorType(mpModel, &mTevStr);
+    mDoExt_modelUpdateDL(mpModel);
+    return TRUE;
 }
 
-/* 00001044-00001064       .text daWall_Create__FPv */
-static s32 daWall_Create(void*) {
-    /* Nonmatching */
+static s32 daWall_Create(void* i_this) {
+    return ((daWall_c*)i_this)->_create();
 }
 
-/* 00001064-00001088       .text daWall_Delete__FPv */
-static BOOL daWall_Delete(void*) {
-    /* Nonmatching */
+static BOOL daWall_Delete(void* i_this) {
+    return ((daWall_c*)i_this)->_delete() & 0xFF;
```

Comment:
```suggestion
    return ((daWall_c*)i_this)->_delete();
```

when a static method returns a `BOOL` and it has a clrlwi like this, it's probably a sign that the inner method (`_delete`, `_draw`, etc) should return a `bool` as that will create the clrlwi automatically. then you don't need to manually `& 0xFF` which is a fakematch.